pkg/controller/kibana/stackmon/sidecar.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stackmon
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	kbv1 "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/defaults"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/stackmon"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/stackmon/monitoring"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/volume"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/kibana/network"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+)
+
+const (
+	// cfgHashLabel is used to store a hash of the Metricbeat and Filebeat configurations.
+	// Using only one label for both configs to save labels.
+	cfgHashLabel = "kibana.k8s.elastic.co/monitoring-config-hash"
+
+	kibanaLogsVolumeName = "kibana-logs"
+	kibanaLogsMountPath  = "/usr/share/kibana/logs"
+	kibanaLogFilename    = "kibana.json"
+)
+
+func Metricbeat(client k8s.Client, kb kbv1.Kibana) (stackmon.BeatSidecar, error) {
+	if !kb.Spec.ElasticsearchRef.IsDefined() {
+		return stackmon.BeatSidecar{}, errors.New("monitoring does not support Kibana without associated Elasticsearch")
+	}
+	associatedEsNsn := kb.Spec.ElasticsearchRef.NamespacedName()
+	if associatedEsNsn.Namespace == "" {
+		associatedEsNsn.Namespace = kb.Namespace
+	}
```

Comment:
I'm surprised we have that check for metricbeat but not for filebeat?

Reply:
The base Filebeat configuration is purely static while the base Metricbeat configuration is a template to inject the credentials of the monitored Elasticsearch/Kibana into it. So, for Metricbeat we need to retrieve the monitoring user/password using this namespaced name. There is no this need for Filebeat.